From 25f9f92eb335338fbdbf36ec41acd7f275c82a4c Mon Sep 17 00:00:00 2001
From: Tomas Zak <tomas.zak@turris.com>
Date: Thu, 30 Jan 2025 18:55:27 +0100
Subject: [PATCH] python-rpds-py: fix compiling on powerpc

Due to multiple definitions of MAP_HUGETLB in cargo libc update libc from
version 0.2.147 to 0.2.148 where it is fixed
---
 ...1-fix-libc-multiple-definitions-powerpc.patch | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)
 create mode 100644 lang/python/python-rpds-py/patches/001-fix-libc-multiple-definitions-powerpc.patch

diff --git a/lang/python/python-rpds-py/patches/001-fix-libc-multiple-definitions-powerpc.patch b/lang/python/python-rpds-py/patches/001-fix-libc-multiple-definitions-powerpc.patch
new file mode 100644
index 000000000..5b1a1d715
--- /dev/null
+++ b/lang/python/python-rpds-py/patches/001-fix-libc-multiple-definitions-powerpc.patch
@@ -0,0 +1,16 @@
+Index: rpds_py-0.10.6/Cargo.lock
+===================================================================
+--- rpds_py-0.10.6.orig/Cargo.lock
++++ rpds_py-0.10.6/Cargo.lock
+@@ -37,9 +37,9 @@ checksum = "bfa799dd5ed20a7e349f3b4639aa
+ 
+ [[package]]
+ name = "libc"
+-version = "0.2.147"
++version = "0.2.148"
+ source = "registry+https://github.com/rust-lang/crates.io-index"
+-checksum = "b4668fb0ea861c1df094127ac5f1da3409a82116a4ba74fca2e58ef927159bb3"
++checksum = "9cdc71e17332e86d2e1d38c1f99edcb6288ee11b815fb1a4b049eaa2114d369b"
+ 
+ [[package]]
+ name = "lock_api"
-- 
2.48.1

